mixer/pkg/perf/controller_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perf
+
+import (
+	"testing"
+)
+
+func TestControllerBasic(t *testing.T) {
+	c, err := newController()
+	if err != nil {
+		t.Fatalf("Error creating controller: %v", err)
+	}
+
+	s, err := NewClientServer(c.location())
```

Comment:
For now, should we just add a function on the controller, "AddClient(name string) and let the controller manager the wait etc for me and I don't need to know there exists a NewClientServer

Reply:
Currently, such logic is located in the "Run" methods. In the future, we can refactor and make the Controller to be a bit more logic-heavy, with figuring out how to launch the clients themselves as well. However, I want to avoid checking a bare minimum infrastructure first, and make improvements on it later.